lib/Containers/SmallVector.h

```diff
@@ -33,6 +33,129 @@ namespace containers {
 template <class T, std::size_t BufSize = 64, std::size_t ElementAlignment = alignof(T)>
 using SmallVector = std::vector<T, detail::short_alloc<T, BufSize, ElementAlignment>>;
 
+// helper class that combines a vector with a small-sized arena.
+// objects of this class are supposed to be created on the stack and are
+// neither copyable nor movable.
+// the interface is a subset of std::vector's interface.
+template <class T, std::size_t BufSize = 64, std::size_t ElementAlignment = alignof(T)>
+class SmallVectorWithArena {
+ public:
+  SmallVectorWithArena() noexcept
+      : _vector{_arena} {
+    // reserve enough room in the arena to avoid early re-allocations later.
+    // note that the initial allocation will take place in the arena, and not
+    // use any extra heap memory.
+    // later allocations can grow the vector beyond the size of the arena, and
+    // then heap allocations will need to be made.
+    _vector.reserve(BufSize / sizeof(T));
+  }
+
+  SmallVectorWithArena(SmallVectorWithArena const& other) = delete;
+  SmallVectorWithArena& operator=(SmallVectorWithArena const& other) = delete;
+  SmallVectorWithArena(SmallVectorWithArena&& other) = delete;
+  SmallVectorWithArena& operator=(SmallVectorWithArena&& other) = delete;
+  
+  using AllocatorType = detail::short_alloc<T, BufSize, ElementAlignment>;
+  using iterator = typename std::vector<T, AllocatorType>::iterator;
+  using const_iterator = typename std::vector<T, AllocatorType>::const_iterator;
+  using reverse_iterator = typename std::vector<T, AllocatorType>::reverse_iterator;
+  using const_reverse_iterator = typename std::vector<T, AllocatorType>::const_reverse_iterator;
+
+  void reserve(std::size_t n) {
+    _vector.reserve(n);
+  }
+  
+  void resize(std::size_t n) {
+    _vector.resize(n);
+  }
+
+  void clear() {
+    _vector.clear();
+  }
+  
+  iterator insert(iterator pos, T const& value) {
+    return _vector.insert(pos, value);
+  }
+  
+  iterator insert(const_iterator pos, T const& value) {
+    return _vector.insert(pos, value);
+  }
+  
+  template <class... Args>
+  void emplace_back(Args&&... args) {
+    _vector.emplace_back(std::forward<Args>(args)...);
+  }
+
+  void pop_back() {
+    _vector.pop_back();
+  }
+  
+  void push_back(T const& value) {
+    _vector.emplace_back(value);
+  }
+
+  std::size_t size() const noexcept {
+    return _vector.size();
+  }
+
```

Comment:
Can we also add `capacity`?
```suggestion
  std::size_t size() const noexcept {
    return _vector.size();
  }

  std::size_t capacity() const noexcept {
    return _vector.capacity();
  }

```